src/test/java/org/mvel2/util/StringAppenderTest.java

```diff
@@ -0,0 +1,109 @@
+package org.mvel2.util;
+
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.mvel2.util.StringAppender;
+
+public class StringAppenderTest {
+
+  @Rule
+  public final ExpectedException thrown = ExpectedException.none();
+
+  @Test
+  public void testCharConstructor() {
+    final StringAppender stringAppender = new StringAppender('a');
+    Assert.assertEquals("", stringAppender.toString());
+    // This looks to be exhibiting a potential bug, shouldn't this return "a"?
+  }
+
+  @Test
+  public void testAppendNull() {
+    final StringAppender stringAppender = new StringAppender("a");
+    String s = null;
+    stringAppender.append(s);
+    Assert.assertEquals("a", stringAppender.toString());
+  }
+
+  @Test
+  public void testAppendByteArr() {
+    final StringAppender stringAppender = new StringAppender("a");
+    stringAppender.append("b".getBytes());
+    stringAppender.append("c".getBytes());
+    Assert.assertEquals("abc", stringAppender.toString());
+  }
+
+  @Test
+  public void testAppendByteArrSub() {
+    final StringAppender stringAppender = new StringAppender("a");
+    stringAppender.append("bcd".getBytes(), 1, 2);
+    stringAppender.append("e".getBytes(), 1, 0);
+    Assert.assertEquals("acd", stringAppender.toString());
+  }
+
+  @Test
+  public void testAppendCharArrSub() {
+    final StringAppender stringAppender = new StringAppender(new char[] { 'a' });
+    stringAppender.append(new char[] { 'b', 'c', 'd' }, 1, 2);
+    stringAppender.append(new char[] { 'e' }, 1, 0);
+    Assert.assertEquals("acd", stringAppender.toString());
+  }
+
+  @Test
+  public void testGetChars() {
+    final StringAppender stringAppender = new StringAppender(new StringBuffer("abc"));
+    Assert.assertArrayEquals(new char[] { 'a', 'b', 'c' }, stringAppender.getChars(0, 3));
+  }
+
+  @Test
+  public void testGetCharsSub() {
+    final StringAppender stringAppender = new StringAppender(new StringBuffer("abcdef"));
+    final char[] target = { 'g', 'h', 'i' };
```

Comment:
Could target be inlined or does it make one of the lines too long?

Reply:
No, because i'm using this array in the assert as well as the method under test.